### ISSUE TITLE
5 fold cross-validation on the training data?

### ISSUE THREAD

**Issue by chengjianhong** [ellisdg/3DUnetCNN#173]
Hi,
The Author of the paper indicates that it uses 5 fold cross-validation during training. However the implementation is only the80% training data and  20% validation data to train. and How to implement the code with 5 fold cross-validation? Can you give me some advice?

Best Wishes,
Jianhong Cheng

**Comment by chengjianhong** [comment]
I make it.

**Comment by akhanss** [comment]
Hi @chengjianhong 
I am just wondering how you have made this. Thanks in advance.
Best,
Azam.

**Comment by Johncheng1** [comment]
Hi @chengjianhong ，can you tell me how to do 5 fold cross-validation on the training data?  Thank you!

**Comment by chengjianhong** [comment]
@Johncheng1 @akhanss this is a training strategy that makes the result have representative ability. your need make the whole training set divide into five parts, and each part should be used as validation set once during training  one model.

**Comment by Johncheng1** [comment]
@chengjianhong  ok，thanks for your answer.

**Comment by zhoutonxgue** [comment]
@chengjianhong  I also want to know about the 5 fold cross-validation, did you change the original code to do the 5 fold cross-validation, and could you please tell me which file you changed, thanks in advance!

**Comment by chengjianhong** [comment]
@zhoutonxgue Yes, you can change the `split_list` function in `generator.py` and do other corresponding changes.

**Comment by zhoutonxgue** [comment]
@chengjianhong  thanks, so split=0.8 in the `split_list` means train: test=8:2? when I run predict.py it gave the 20% data's prediction which is just the validation_list, I was confused, it means the validation is the test?  
I have another question, when I run train_isensee2017.py I got the error 'Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)' I can fix it by delete the validation_data=validation_generator in function train model in training.py but then I can't get val_loss, it's strange. Could you please give me some suggestions? Thank you very much.

**Comment by chengjianhong** [comment]
@zhoutonxgue `split=0.8 ` means it make the whole data split into 80% data for training and 20% data for validation. validation is necessary to tune parameters during training model. As for your another question, I have never met it.

**Comment by zhoutonxgue** [comment]
@chengjianhong  ok, thanks, and the 20% validation data is also used to predict？

**Comment by akhanss** [comment]
> @Johncheng1 @akhanss this is a training strategy that makes the result have representative ability. you need make the whole training set divide into five parts, and each part should be used as validation set once during training one model. Finally, average the results from five models.

Thnx for your kind reply. I am just wondering if you can share the code snippet of how you do average the results from five models. Thanks in advance.
Regards,
Azam.

**Comment by kylinJo** [comment]
> @zhoutonxgue Yes, you can change the `split_list` function in `generator.py` and do other corresponding changes.

can you tell me how to add  5 fold cross-validation?
